### PR TITLE
[DOC-3030] [SMP-2398] [PL-46414] SMP pre-flight validation plugin

### PR DESCRIPTION
**Pull request by brian-f-harness**
SMP pre-flight validation plugin

[Run a pre-flight validation plugin](https://650b03e1eeddf1008aa64058--harness-developer.netlify.app/docs/self-managed-enterprise-edition/troubleshooting/run-preflight-validation-plugin)

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — APPROVED** [review]

File: docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+title: Run a pre-flight validation plugin
+description: Learn how to run a pre-flight validation plugin to perform pre-install, pre-upgrade, post-install, and post-upgrade checks on your setup. 
+# sidebar_position: 5
+---
+
+The pre-flight validation plugin allows you to perform pre-install, pre-upgrade, post-install, and post-upgrade checks on your setup. This optional plugin assists with installations and upgrades for Harness Self-Managed Enterprise Edition. You can use the plugin to identify potential issues during deployment and ensure that all requirements are met before the deployment process begins. Suggestions are provided if the requirements are not met and the validation checks fail. You can also use the plugin architecture to build new validation checks for your custom setup.
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md:7]:
```suggestion
The pre-flight validation plugin allows you to perform pre-install, pre-upgrade, post-install, and post-upgrade checks on your setup. This optional plugin assists with installations and upgrades for Harness Self-Managed Enterprise Edition. You can use the plugin to identify potential issues during deployment and ensure that all requirements are met before the deployment process begins. Suggestions are provided if the requirements are not met and the validation checks fail.
```

File: docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+Helm plugins are installed using the `helm plugin install <path|url>` command. You can pass in a path to a plugin on your local file system or a URL of a remote VCS repo. The `helm plugin install` command clones or copies the plugin at the path/URL in `$HELM_PLUGINS`.
+
+To install the plugin using the GitHub repository, run the following command:
+
+```
+helm plugin install git@github.com:harness/Plugins-helm.git
+```
+
+Alternatively, for an [airgapped setup](/docs/self-managed-enterprise-edition/self-managed-helm-based-install/install-in-an-air-gapped-environment/), you can set the path to the directory where the plugin is installed on your system:
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md:24]:
```suggestion
Alternatively, for an [airgapped setup](/docs/self-managed-enterprise-edition/self-managed-helm-based-install/install-in-an-air-gapped-environment/), you can set the path to the directory where the plugin is downloaded on your system:
```

File: docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+First, download validator plugin tar file from the Harness Google Container Registry (GCR), and then run the following command to install the plugin:
```
Comment (kapilgarg1996) [inline docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md:26]: We need to provide the link for container registry

Comment (brian-f-harness) [reply docs/self-managed-enterprise-edition/self-managed-helm-based-install/run-preflight-validation-plugin.md:26]: @kapilgarg1996 Thanks for your review. I committed the suggestions. Do you have the link to the tar file in the Harness GCR?

**Review by pratmit — DISMISSED** [review]
LGTM

**Comment by brian-f-harness** [comment]
Pending public repo per @kapilgarg1996. Will merge after repo is public.

**Comment by ravilach** [comment]
@brian-f-harness do we still need this? Was flagged as a stale branch. Thanks.

**Comment by brian-f-harness** [comment]
> @brian-f-harness do we still need this? Was flagged as a stale branch. Thanks.

@ravilach Yes, we still need it. The SMP team needs to make the repo public before we merge.

**Comment by brian-f-harness** [comment]
Closing for now. Will open a new PR with this content after the repo is made public.

**Pull request closed without merging**